versioned_docs/version-V3/guides/providing-liquidity/collect-fees.md

```diff
@@ -0,0 +1,69 @@
+---
+id: collect-fees
+title: Collecting Fees
+sidebar_position: 3
+---
+
+## Collecting Fees
+
+Make sure to go through the [first guide](./setting-up-your-contract.md) before continuing to this section
+
+To collect the fees of an owner position, we will need to transfer the NFT from the calling address, assign the relevant variables from the NFT to local variables within our function, and pass those variables to the `nonfungiblePositionManager` when we call `collect`.
+
+- Note the contract must have custody the NFT to be able to collect the fees
+
+This function specifically collects all fees and sends to original owner of the NFT while still custodying the NFT
+Set `amount0Max` and `amount1Max` to type(uint128).max to collect all fees
+
+```solidity
+    /// @notice Collects the fees associated with provided liquidity
+    /// @dev The contract must hold the erc721 token before it can collect fees
+    /// @param tokenId The id of the erc721 token
+    /// @return amount0 The amount of fees collected in token0
+    /// @return amount1 The amount of fees collected in token1
+    function collectAllFees(uint256 tokenId) external returns (uint256 amount0, uint256 amount1) {
+        // Caller must own the ERC721 position
+        // Call to safeTransfer will trigger `onERC721Received` which must return the selector else transfer will fail
+        nonfungiblePositionManager.safeTransferFrom(msg.sender, address(this), tokenId);
+
+        // set amount0Max and amount1Max to uint256.max to collect all fees
+        // alternatively can set recipient to msg.sender and avoid another transaction in `sendToOwner`
+        INonfungiblePositionManager.CollectParams memory params =
+            INonfungiblePositionManager.CollectParams({
+                tokenId: tokenId,
+                recipient: address(this),
+                amount0Max: type(uint128).max,
+                amount1Max: type(uint128).max
+            });
+
+        (amount0, amount1) = nonfungiblePositionManager.collect(params);
+
+        // send collected feed back to owner
+        _sendToOwner(tokenId, amount0, amount1);
+    }
+```
+
+## Sending Fees To The Calling Address
+
+This internal function takes the fees due amounts, which were previously populated in the last function, as arguments to `safeTransfer` which sends the fees which were previously collected to our contract to `owner`.
```

Comment:
Let's specify here that this is a helper function we need when we call collect fees. Any time we need to send profits (fees) or tokens back to the owner we can call _sendToOwner